src/lib/cache/TTLCache.js

```diff
@@ -33,11 +43,23 @@ class TTLCache<K, V> {
       expire,
     };
 
+    while (this.cache.size >= this.options.max) {
+      this.shift();
+    }
+
     this.cache.set(key, item);
 
     return true;
   }
 
+  shift() {
+    const key = this.cache.keys().next().value;
```

Comment:
I'm not sure – do have ES6 maps have a specified order? If not, this could remove the last inserted item!

Reply:
Yes, they have: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Objects_and_maps_compared

> The keys in Map are ordered while keys added to object are not. Thus, when iterating over it, a Map object returns keys in order of insertion.

Reply:
Super :)

Reply:
I will add a comment, to clarify this for future readers. ;)